process-engine/skeleton
# Feature/execute process

## What did you change?

update the dependency-versions so that `executeProcess` can be used

## How can others test the changes?

- clone the skeleton
- `npm install`in the folders `process-engine-browser` and `process-engine-server`
- `npm run build` in the `process-engine-browser`-folder
- `npm start` in the folders `process-engine-browser` and `process-engine-server`
- create a process where the process is forced to be executed in a browser-instance
  ```
  feature: [{"name": "Name", "value": "Skeleton_Server_Browser"}]
  ```

- add a script-task with inline-script that `console.log`s and returns something:

  ```javascript
  console.log('test');
  return {test: '456'};
  ```

  ![bildschirmfoto 2017-11-20 um 12 58 43](https://user-images.githubusercontent.com/20770029/33017690-916da8ee-cdf3-11e7-9aac-f5f48ef1c105.png)

- open multiple browser-instances on `http://localhost:9000`
- open postman and send an `executeProcess`-command  to the node-instance:
  ```
  url: http://localhost:8000/processengine/execute/?key=hallo
  content-type: application/json
  body: {}
  ```
- run the request multiple times and see, how every time exactly one of the browser-instances executes the script task, and the result of the http-request is always the result you returned in the script-task

![ezgif com-optimize](https://user-images.githubusercontent.com/20770029/33020107-3e18705c-cdfd-11e7-8944-2ce0d157ead3.gif)


## PR-Checklist

Please check the boxes in this list after submitting your PR:
